[#3] API 문서 작성

sssukho opened the pull request:
1. Shop_API_문서.md
   - 미용 시술소와 관련된 API들의 스펙을 정의한 문서입니다.
2. Category_API_문서.md
   - 카테고리와 관련된 API들의 스펙을 정의한 문서입니다.
3. Reservation_API_문서.md
   - 예약과 관련된 API들의 스펙을 정의한 문서입니다.
4. Review_API_문서.md
   - 리뷰(시술 후기)와 관련된 API들의 스펙을 정의한 문서입니다.
5. Member_API_문서.md
   - 회원과 관련된 API들의 스펙을 정의한 문서입니다.

추가될 수 있는 도메인(ex. statistics, notification)과 관련된 API들에 대한 문서는 차후 작성할 예정입니다.

sssukho committed:
docs: Shop API 문서 작성

sssukho committed:
docs: Category API 문서 작성 및 Shop API 문서 내 일부 오타 수정

sssukho committed:
docs: Reservation API 문서 작성 및 기존 문서 내 응답 형태 수정

sssukho committed:
docs: Review API 문서 작성

f-lab-lyan commented on .gitignore:
왜 이파일을 다시 수정하셨나요? 그냥 심플하게 .idea만 하셔되 될 것 같은데.. :-)

sssukho replied:
해당 파일 수정된 이유는 [멀티 모듈 생성 작업](https://github.com/f-lab-edu/beautify-project/issues/1) 당시 테스트 용도로 수정을 했었는데 섞여서 push가 되어서 그렇습니다 😂

이전에 작업했던 gitignore 파일을 사용할 예정입니다!

f-lab-lyan commented on docs/Category_API_문서.md:
```diff
@@ -0,0 +1,471 @@
+  "description": "헤어라인모발이식, 머리모발이식, 이마축소, 두피문신 등의 시술이 포함된 카테고리 입니다.",
+  "operations": [
+    {
+      "id": "ace52d35",
```
카테고리를 등록하기 전에 시술들은 사전에 등록되어 있어야 하는건가요?

sssukho replied:
넵 맞습니다.

시술 같은 경우, 시술소 사장 권한의 사용자가 무분별하게 만들면 안되기 때문에 정책적으로 미리 생성된 시술들에 한해서만 카테고리에  할당될 수 있도록 계획했습니다.

f-lab-lyan commented on docs/Category_API_문서.md:
```diff
@@ -0,0 +1,471 @@
+
+| 헤더 key      | 필수 여부 | 설명                                  |
+| ------------- | --------- | ------------------------------------- |
+| Authorization | O         | Bearer ${ACCESS_TOKEN} |
```
카테고리 요청도 로그인이 되어있어야 하는 것인가요?

sssukho replied:
비회원 케이스에 대해서 아예 생각을 안했습니다 😅
카테고리 요청 뿐만 아니라, Shop 쪽이나 Review 쪽도 수정하겠습니다.

f-lab-lyan commented on docs/Reservation_API_문서.md:
```diff
@@ -0,0 +1,374 @@
+    "id": "ace52d35-9ee8-49fb-93af-27195c1e0519",
+    "name": "두피문신"
+  },
+  "user": {
```
유저정보를 별도로 받는건 ACCESS_TOKEN으로 가져오는 정보와 다른 정보로도 예약을 가능하게 하기 위함인가요?

sssukho replied:
아닙니다. 
단순 사용자 정보를 찾기 위함이었는데, 액세스 토큰이 있는 경우에는 액세스 토큰에서 사용자 정보를 추출할 수 있으니 필요 없는 데이터 인 것 같습니다.

수정하겠습니다!

sssukho committed:
docs: 불필요 및 중복 정보 삭제

- Shop, Review, Category, Reservation 조회시 불필요 정보(access token) 삭제
- Reservation 등록 요청시 요청 메시지 본문에 사용자 정보 중복으로 인한 삭제 (access token 으로 대체 가능)

sssukho committed:
Merge branch 'main' into feature/3
